### ISSUE TITLE
Fails to export with autobatching

### ISSUE THREAD

**Issue by Carlson-J** [ultralytics/ultralytics#1680]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Export

### Bug

When the batch size is set to -1 it fails to export.

```
Traceback (most recent call last):
  File "train.py", line 27, in <module>
    model.export(device=0, dynamic=True, half=True)
  File "ultralytics/yolo/engine/model.py", line 298, in export
    return Exporter(overrides=args)(model=self.model)
  File "torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "ultralytics/yolo/engine/exporter.py", line 174, in __call__
    im = torch.zeros(self.args.batch, 3, *self.imgsz).to(self.device)
RuntimeError: Trying to create tensor with negative dimension -1: [-1, 3, 1024, 1024]
```

### Environment

Ultralytics YOLOv8.0.55 🚀 Python-3.10.4 torch-1.13.1+cu117

### Minimal Reproducible Example

model = ImsarModel('yolov8n.pt')
model.train(data='./configs/data.yaml', batch=-1)
model.export()

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Carlson-J this is user error, not a bug. AutoBatch only applies to training. What batch size would you want to be computed for an exported model, I don't understand what you're seeking with this argument value.

**Comment by Carlson-J** [comment]
If I train the model and then export it the batch size stays at -1. It seems like it should change to what the auto batching set it to so that I don't get an error in this case. I was using `model.export(dynamic=True)`, since the batch size may change runtime.

**Comment by glenn-jocher** [comment]
@Carlson-J ah yes, my mistake. I see your export argument inherited the original training batch=-1 command. I think for export we probably want to not remember training batch size as the two batch sizes are unrelated. I'll add a TODO to prioritize a fix here.

**Comment by glenn-jocher** [comment]
@Carlson-J circling back to this. I'm able to reproduce with

```python
from ultralytics import YOLO

model = YOLO()
model.train(data='coco8.yaml', imgsz=64, epochs=3, batch=-1)
model.export(format='onnx')
```

Working on a fix.

**Comment by glenn-jocher** [comment]
@Carlson-J Great news 😃! Your original issue may now be resolved ✅ in PR #3217. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
